packages/venia-concept/src/components/ProductImageCarousel/thumbnail.js

```diff
@@ -20,6 +20,12 @@ class Thumbnail extends Component {
         itemIndex: PropTypes.number,
         onClickHandler: PropTypes.func.isRequired
     };
+    // TODO: When we implement hooks/mergeClasses from #1078 we can replace this
+    // with a useWindowSize hook and rerender when the width changes across the
+    // breakpoint. This is important when changing from portrait to landscape.
+    isDesktop = () => {
+        return window.innerWidth > 640;
```

Comment:
The media rule to display thumbnails is `@media (min-width: 1024px)`, so this still would request images and not display for widths between 641-1023.

```suggestion
        return window.innerWidth >= 1024;
```

Reply:
I think there is some confusion around this component. The thumbnails are actually dual mode. When the window width is > 640 we display actual thumbnail images. when the window is less than 640 we still display a thumbnail component but it's just the icons below the image:

[![Image from Gyazo](https://i.gyazo.com/d743ffc0ef20db6d357ca3a7599ca577.png)](https://gyazo.com/d743ffc0ef20db6d357ca3a7599ca577)

Reply:
A more long term solution that I don't think we should get into here would be to actually refactor these into separate components.

Reply:
Ok I was wrong -- 1024 is correct, although we still should probably separate these out eventually, IMO :)